source/MaterialXCore/Node.cpp

```diff
@@ -411,9 +411,15 @@ vector<ElementPtr> GraphElement::topologicalSort() const
         size_t connectionCount = 0;
         for (size_t i = 0; i < child->getUpstreamEdgeCount(); ++i)
         {
-            if (child->getUpstreamEdge(i))
+            Edge upstreamEdge = child->getUpstreamEdge(i);
+            if (upstreamEdge)
             {
-                connectionCount++;
+                ElementPtr elem = upstreamEdge.getUpstreamElement()->getParent();
```

Comment:
Since `topologicalSort` only applies to the nodes within a single graph, connections to nodes in a different graph need to be omitted from consideration in the sort.